flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkServiceTest.java

```diff
@@ -1315,6 +1316,36 @@ public void testBlockingDeletionDeleteCallErrorHandling() {
         assertTrue(remaining.toMillis() < 1000);
     }
 
+    @Test
+    public void listingJobExceptionsIsCompatibleWihFlinkV1_17Test() throws Exception {
+        JobExceptionsInfoWithHistory exceptionInfoWithNullFailureLabels =
+                new JobExceptionsInfoWithHistory(
+                        new JobExceptionsInfoWithHistory.JobExceptionHistory(
+                                java.util.Collections.singletonList(
+                                        new JobExceptionsInfoWithHistory.RootExceptionInfo(
+                                                "org.apache.flink.util.FlinkExpectedException",
+                                                "The TaskExecutor is shutting down.",
+                                                1755998006623L,
+                                                null,
+                                                "Source: Events Generator Source (2/2) - execution #0",
+                                                "10.244.0.105:44401",
+                                                "basic-example-taskmanager-1-1",
+                                                java.util.Collections.emptyList())),
+                                false));
```

Comment:
It would be nice to take a concrete exception JSON from an 1.17 job instead and test that. This test doesn't actually guarantee that it will work with 1.17

Reply:
Makes sense, thanks! I changed it to json-based test